src/Sulu/Component/Content/Compat/Structure.php

```diff
@@ -763,17 +763,9 @@ public function setInternal($internal)
      */
     public function getNodeName()
     {
-        if (
-            self::NODE_TYPE_INTERNAL_LINK === $this->getNodeType() &&
-            null !== $this->getInternalLinkContent() &&
-            $this->getInternalLinkContent()->hasProperty('title')
-        ) {
-            return $this->internalLinkContent->getPropertyValue('title');
-        } elseif ($this->hasProperty('title')) {
-            return $this->getPropertyValue('title');
+        if ($this->hasProperty('title')) {
```

Comment:
internalLinkContent don't exist so it will always get into the `$this->getPropertyValue('title');` 
/cc @danrot maybe you know more about this internalLink case?